components/org.wso2.carbon.identity.conditional.auth.functions.elk/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/elk/util/ElasticConfigProvider.java

```diff
@@ -0,0 +1,112 @@
+    private static final String ES_QUERY_PARAM_USERNAME = "[ES_QUERY_PARAM_USERNAME]";
+    private static final String ES_QUERY_PARAM_DURATION = "[ES_QUERY_PARAM_DURATION]";
+    private static final String ES_QUERY_PARAM_DURATION_DEFAULT = "5m";
+    private static final String ES_QUERY_PARAM_LOGIC = "[ES_QUERY_PARAM_LOGIC]";
+    private static final String ES_QUERY_PARAM_LOGIC_DEFAULT = "state.sum.add(doc['amount'].value);";
+    private static final String ES_QUERY_PARAM_THRESHOLD = "[ES_QUERY_PARAM_THRESHOLD]";
+    private static final String ES_QUERY_PARAM_THRESHOLD_DEFAULT = "10000";
+    private static final String ES_INDEX_DEFAULT = "transaction";
+
+    private static final ElasticConfigProvider instance = new ElasticConfigProvider();
+
+    private ElasticConfigProvider() {
+
+    }
+
+    public static ElasticConfigProvider getInstance() {
+
+        return instance;
+    }
+
+    /**
+     *
+     * @param elasticDomain Elasticsearch host domain
+     * @param params User defined parameter from script
+     * @return Elasticsearch query URL
+     */
+    public String getElasticSearchUrl(String elasticDomain, Map<String, String> params) {
+
+        return elasticDomain + params.getOrDefault("index", ES_INDEX_DEFAULT) + "/_search?size=0";
+    }
+
+    /**
+     *
+     * @param params User defined parameter from script
+     * @return Elasticsearch query to get risk score
+     * @throws IOException For query file read failures
+     */
+    public String getQuery(Map<String, String> params) throws IOException {
+
+        String query = readConfigFile();
+        query = query
+                .replace(
+                        ES_QUERY_PARAM_USERNAME, params.get("username")
+                ).replace(
+                        ES_QUERY_PARAM_DURATION,
+                        params.getOrDefault("duration", ES_QUERY_PARAM_DURATION_DEFAULT)
+                ).replace(
+                        ES_QUERY_PARAM_LOGIC,
+                        params.getOrDefault("riskLogic", ES_QUERY_PARAM_LOGIC_DEFAULT)
+                ).replace(
+                        ES_QUERY_PARAM_THRESHOLD,
+                        params.getOrDefault("threshold", ES_QUERY_PARAM_THRESHOLD_DEFAULT)
+                );
+
+        return query;
+    }
+
+    private String readConfigFile() throws IOException {
+
+        String fileContent;
+        InputStream inputStream = getClass().getClassLoader().getResourceAsStream(CONFIG_PATH);
+
+        if (inputStream == null) {
+            fileContent = "FILE READ FAILED";
+        } else {
+            InputStreamReader isReader = new InputStreamReader(inputStream);
+            BufferedReader reader = new BufferedReader(isReader);
+            StringBuilder sb = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                sb.append(line);
+            }
+            fileContent = sb.toString();
+        }
+
+        return fileContent;
+    }
+
```

Comment:
Remove extra line.

Reply:
Resolved in https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/110